File: server/datasets/java_source-code/Multiple-C1/mixed_attack_1.java
/*
 * Graph BFS - Mixed Attack Submission
 * Derived from organic_4.java. Combines identifier renaming AND statement reordering
 * (helper function moved below main logic, loop body lines swapped).
 */

import java.util.*;

class Solution {
    public static List<Integer> bfsOverMatrix(int[][] adjMatrix, int originIndex) {
        int totalNodes = adjMatrix.length;
        boolean[] seenFlags = new boolean[totalNodes];
        List<Integer> traversalOrder = new ArrayList<>();
        Queue<Integer> pending = new ArrayDeque<>();

        seenFlags[originIndex] = true;
        pending.add(originIndex);

        while (!pending.isEmpty()) {
            int active = pending.poll();

            for (int targetIndex = 0; targetIndex < totalNodes; targetIndex++) {
                if (adjMatrix[active][targetIndex] == 1 && !seenFlags[targetIndex]) {
                    pending.add(targetIndex);
                    seenFlags[targetIndex] = true;
                }
            }

            traversalOrder.add(active);
        }
        return traversalOrder;
    }

    public static List<String> labelLookup(List<Integer> indices, String[] labels) {
        List<String> result = new ArrayList<>();
        for (int i : indices) {
            result.add(labels[i]);
        }
        return result;
    }

    public static void main(String[] args) {
        String[] labels = {"A", "B", "C", "D", "E"};
        int[][] adjMatrix = {
            {0, 1, 1, 0, 0},
            {1, 0, 0, 1, 0},
            {1, 0, 0, 0, 1},
            {0, 1, 0, 0, 1},
            {0, 0, 1, 1, 0}
        };
        List<Integer> order = bfsOverMatrix(adjMatrix, 0);
        System.out.println(labelLookup(order, labels));
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_c1/mixed_attack_1.py
"""
Graph BFS - Mixed Attack Submission
Derived from organic_4.py (adjacency matrix BFS). Combines two evasion
techniques at once: identifier renaming AND statement reordering
(helper function moved below main logic, loop body lines swapped).
"""

from collections import deque


def bfs_over_matrix(adj_matrix, origin_index):
    total_nodes = len(adj_matrix)
    seen_flags = [False] * total_nodes
    traversal_order = []
    pending = deque([origin_index])
    seen_flags[origin_index] = True

    while pending:
        active = pending.popleft()

        for target_index in range(total_nodes):
            if adj_matrix[active][target_index] == 1 and not seen_flags[target_index]:
                pending.append(target_index)
                seen_flags[target_index] = True

        traversal_order.append(active)

    return traversal_order


def label_lookup(indices, labels):
    return [labels[i] for i in indices]


if __name__ == "__main__":
    labels = ["A", "B", "C", "D", "E"]
    adj_matrix = [
        [0, 1, 1, 0, 0],
        [1, 0, 0, 1, 0],
        [1, 0, 0, 0, 1],
        [0, 1, 0, 0, 1],
        [0, 0, 1, 1, 0],
    ]
    order = bfs_over_matrix(adj_matrix, 0)
    print(label_lookup(order, labels))
